source/grid/grid_generator.cc

```diff
@@ -3927,6 +3927,96 @@ namespace GridGenerator
   }
 
 
+  template <int dim, int spacedim>
+  void replicate_triangulation(const Triangulation<dim, spacedim> &basic_tria,
+                               const std::array<Point<spacedim>, 2> &diagonals,
+                               const std::array<unsigned int, dim> &repetitions,
+                               Triangulation<dim, spacedim> &result)
+  {
```

Comment:
Do we need to provide diagonals at all? It would just be another loop through all the cells to find them here automatically.

Reply:
That could be done for sure, but then that's with an underlying assumption that the basic triangulation is with small size to keep looping over cells trivial. Do we assume this?

Reply:
In comparison to everything else we are doing here, it should still be cheap.